avm/res/dashboard/grafana/tests/e2e/max/main.test.bicep

```diff
@@ -0,0 +1,127 @@
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-dashboard.grafana-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param resourceLocation string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'crmax'
+
+@description('Optional. A token to inject into the name of each resource.')
+param namePrefix string = 'test' //'#_namePrefix_#'
+
+// ============ //
+// Dependencies //
+// ============ //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: resourceLocation
+}
+
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-nestedDependencies'
+  params: {
+    managedIdentityName: 'dep-${namePrefix}-msi-${serviceShort}'
+    location: resourceLocation
+    virtualNetworkName: 'dep-${namePrefix}-vnet-${serviceShort}'
+  }
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [
+  for iteration in ['init', 'idem']: {
+    scope: resourceGroup
+    name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
+    dependsOn: [
+      nestedDependencies
+    ]
+    params: {
+      name: '${namePrefix}${serviceShort}001'
+      location: resourceLocation
+      tags: {
+        test: 'large-parameter-set'
+      }
+      apiKey: 'Enabled'
+      autoGeneratedDomainNameLabelScope: 'TenantReuse'
+      deterministicOutboundIP: 'Enabled'
+      enableTelemetry: true
+      enterpriseConfigurations: {}
+      grafanaConfigurations: {
+        smtp: {
+          enabled: false
+          fromAddress: '0.0.0.0'
+          fromName: 'learn.microsoft.com'
+          host: 'learn.microsoft.com'
+          password: 'Learn@123Micro'
+          skipVerify: false
+          user: 'admin'
+          startTLSPolicy: 'MandatoryStartTLS'
+        }
+      }
+      skuName: 'Standard'
+      lock: {
+        kind: 'None'
+        name: 'locknone'
+      }
+      publicNetworkAccess: 'Enabled'
+      privateEndpoints: [
+        {
+          privateDnsZoneResourceIds: [
+            nestedDependencies.outputs.privateDNSZoneResourceId
+          ]
+          subnetResourceId: nestedDependencies.outputs.subnetResourceId
+          roleAssignments: [
+            {
+              roleDefinitionIdOrName: inbuiltRole.builtInRoleNames['Grafana Admin']
+              principalId: nestedDependencies.outputs.managedIdentityPrincipalId
+              principalType: 'ServicePrincipal'
+            }
+            {
+              roleDefinitionIdOrName: inbuiltRole.builtInRoleNames['Grafana Editor']
```

Comment:
The idea is actually that you don't need to do this, but that the user can provide the name because the module itself can resolve it (e.g., `roleDefinitionIdOrName: 'Grafana Editor'`

Reply:
Also - why would you assign grafana roles to the Private Endpoint? The module itself should implement the role assignment interface where it would absolutely make sense to provide these roles. But for the PE alone, I'd recommend to rather set a role like reader